packages/zod/src/index.ts

```diff
@@ -130,17 +132,24 @@ export const generateZodValidationSchemaDefinition = (
       }
 
       if (schema.format === 'date') {
+        if (!context.output.override.useDates) {
+          functions.push([type as string, undefined]);
+        }
         functions.push(['date', undefined]);
         break;
       }
 
-      functions.push([type as string, undefined]);
-
       if (schema.format === 'date-time') {
-        functions.push(['datetime', undefined]);
+        if (!context.output.override.useDates) {
+          functions.push([type as string, undefined], ['datetime', undefined]);
+        } else {
+          functions.push(['date', undefined]);
+        }
         break;
       }
 
+      functions.push([type as string, undefined]);
+
```

Comment:
looks good to me.
and if i had to say, it can makes this simple to handle `useDates` first like:

```ts
      if (
        context.output.override.useDates &&
        (schema.format === 'date' || schema.format === 'date-time')
      ) {
        functions.push(['date', undefined]);
        break;
      }

      functions.push([type as string, undefined]);

      if (schema.format === 'date') {
        functions.push(['date', undefined]);
        break;
      }

      if (schema.format === 'date-time') {
        functions.push(['datetime', undefined]);
        break;
      }
```